### DISCUSSION TITLE
[Question][Hw4] Use of other packages

### DISCUSSION THREAD

**Discussion by pingpingy1** [prosyslab-classroom/cs424-program-reasoning#337] [Q&A]
Is it allowed to use other packages available on Opam?
I plan on using a module that implements the priority queue,
but I was not sure if such was allowed or prohibited.
Should I implement the priority queue myself if I need such features?

Thank you.

**Comment by KihongHeo** [comment]
Hi.

Sorry, but we cannot allow students to install arbitrary packages for auto-grading and Gradescope competitions.

I think you can easily implement a priority queue. One of the simplest ways is to just use the [Set module](https://v2.ocaml.org/api/Set.Make.html) in the OCaml standard lib. 
- You can specify an order of elements when making your module (i.e., argument module of the Make functor. 
- `min_elt` or `max_elt` will give you the minimum or maximum element.

**Comment by pingpingy1** [comment]
Oh, okay.
Thank you very much.
